python_modules/dagster/dagster/_core/definitions/scoped_resources_builder.py

```diff
@@ -78,28 +84,41 @@ def build(self, required_resource_keys: Optional[AbstractSet[str]]) -> Resources
             for key in required_resource_keys
             if key in self.resource_instance_dict
         }
+        processed_resource_instance_dict = {
```

Comment:
Here we get all the bare values from `IAttachBareObjectToContext` resources, but we keep around the raw resource objects so that the op/sensor/schedule invocation code can splat them into kwargs.

Reply:
I think we should use the terms "original resource" objects to describe the ones passed into Definitions and then call `processed_resource_instance_dict` `resources_to_attach_to_context`. Just in these cases be as specific as possible.